backend/inspirehep/snow/api.py

```diff
@@ -0,0 +1,571 @@
+            "Content-Type": "application/json",
+        }
+
+    def login(self):
+        login_payload = {
+            "grant_type": "client_credentials",
+            "audience": "snow-api-sso-protected",
+            "client_id": self.client_id,
+            "client_secret": self.secret,
+        }
+        headers = {
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+        try:
+            response = requests.post(self.auth_url, data=login_payload, headers=headers)
+            response.raise_for_status()
+            self.token = response.json()["access_token"]
+            return True
+        except requests.exceptions.RequestException:
+            raise SnowAuthenticationError
+
+    @relogin_if_needed
+    def create_ticket(self, endpoint, payload):
+        """Creates a ticket
+        Args:
+            endpoint (str): an endpoint name
+            payload (dict): ticket payload
+        """
+        response = requests.post(
+            f"{self.url}/{endpoint}",
+            data=orjson.dumps(payload),
+            headers=self.headers,
+        )
+        response.raise_for_status()
+        return get_response_result(response)["sys_id"]
+
+    @relogin_if_needed
+    def get_ticket(self, ticket_id, params=None):
+        """Get ticket by id
+        Args:
+            params (str|dict): params to be passed to the request query string
+        """
+        if not params:
+            params = {}
+        ticket_response = requests.get(
+            f"{self.url}/u_request_fulfillment/{ticket_id}",
+            params=params,
+            headers=self.headers,
+        )
+        ticket_response.raise_for_status()
+        return get_response_result(ticket_response)
+
+    @relogin_if_needed
+    def get_functional_category(self, category_id):
+        """Get functional categoty by id
+        Args:
+            category_id: SNOW sys_id of the category
+         Returns:
+             human-readable category name (str)
+             None if the category is not found
+        """
+        try:
+            category = (
+                self.get_functional_categories_from_cache()
+                .get(category_id, {})
+                .get("sys_name")
+            )
+            if not category:
+                category = self.get_functional_categories_from_cache(force_update=True)[
+                    category_id
+                ].get("sys_name")
+            return category
+        except KeyError:
+            LOGGER.warning(
+                "Functional category was not found!", category_id=category_id
+            )
+
+    @relogin_if_needed
+    def get_user(self, user_id):
+        """Get user by id
+        Args:
+            user_id (str): id of the user
+        Returns:
+            User name (str) if the user exists
+            None if the user doesn't exist
+        """
+        try:
+            user = self.get_users_from_cache().get(user_id, {}).get("user.name")
```

Comment:
the key is `user.name`?

Reply:
yes :| it's `{"user.name": "Test Name"}`